data/team2020clinical/team2020clinical-extraction.md

```diff
@@ -0,0 +1,89 @@
+team2020 = team2020.merge(patient_df, on='PatientID', how='left')
+
+participants = []
+
+# Group by participant and extract measurements
+
+for patient_id, group in team2020.groupby('PatientID'):
+    # Check if both 'Age' and 'Sex' are not NaN
+    if pd.notna(group['Age'].iloc[0]) and pd.notna(group['Sex'].iloc[0]):
+        participant = {
+            'attributes': {
+                'age': int(group['Age'].iloc[0]),
+                'sex': group['Sex'].iloc[0]
+            },
+            'measurements': []
+        }
+    else:
+        # If age or sex is NaN, exclude 'attributes' from the participant dictionary
+        participant = {
+            'measurements': []
+        }
+
+    for _, row in group.iterrows():
+        if row['value'] == 1:
+            value = "negative"
+        else:
+            value = row['value']
+        measurementN = {
+            'analyte': 'naso_swab_SARSCoV2_N',
+            'time': row['Day'],
+            'value': value
+        }
+        participant['measurements'].append(measurementN)
+        
+    participants.append(participant)
+```
+
+Finally, the data is formatted and output as a YAML file.
+
+```python
+team2020clinical = dict(title="Clinical and virologic characteristics of the first 12 patients with coronavirus disease 2019 (COVID-19) in the United States",
+               doi="10.1038/s41591-020-0877-5",
+               description=folded_str('Respiratory, stool, serum, and urine specimens were submitted for SARS-CoV-2 real-time reverse-transcription polymerase chain reaction (rRT-PCR) testing, viral culture, and whole-genome sequencing. Only nasopharyngeal samples were included in this dataset. Data for the nasopharyngeal swab results were obtained from the combined dataset in the supplementary materials of Challenger et al. (2022), while attributes of hospitalized patients were obtained from the original paper.\n'),
+               analytes=dict(naso_swab_SARSCoV2_N=dict(description=folded_str("SARS-CoV-2 RNA genome copy concentration in nasopharyngeal swab samples. Viral load concentrations were obtained from the combined dataset in the supplementary materials of Challenger et al. (2022)\n"),
+                                        limit_of_quantification="unknown",
+                                        limit_of_detection="unknown",
+                                        specimen="nasopharyngeal_swab", 
+                                        biomarker="SARS-CoV-2", 
+                                        gene_target="N", 
+                                        unit="gc/mL",
+                                        reference_event="symptom onset",)),
+                participants = participants
+                                        )
+
+with open("team2020clinical.yaml","w") as outfile:
+    outfile.write("# yaml-language-server: $schema=../.schema.yaml\n")
+    yaml.dump(team2020clinical, outfile, default_flow_style=False, sort_keys=False)
+outfile.close() 
```

Comment:
Till told us that this `outfile.close()` is not needed with `with open`. I will push a commit to remove this.